app/observers/payment_observer.rb

```diff
@@ -18,6 +18,7 @@ def from_chargeback_to_paid(payment)
   end
 
   def from_pending_refund_to_refunded(payment)
+    return if Contribution.where.not(donation:nil).where(id: payment.contribution).exists?
```

Comment:
I have just realised all donations will go from paid to refunded directly, so we don't need this check at all :)